crates/ra_hir/src/path.rs

```diff
@@ -74,6 +80,26 @@ impl Path {
                     let segment = PathSegment { name: name.as_name(), args_and_bindings: args };
                     segments.push(segment);
                 }
+                ast::PathSegmentKind::Angle { type_ref, trait_ref } => {
+                    assert!(path.qualifier().is_none()); // this can only occur at the first segment
+
+                    // <T as Trait<A>>::Foo desugars to Trait<Self=T, A>::Foo
```

Comment:
Let's add a TODO for `<()>::default` syntax.

Reply:
I kind of forgot about that syntax!